lib/screens/app/profile/profile.dart

```diff
@@ -1,10 +1,223 @@
+import 'package:image_picker/image_picker.dart';
+import 'package:provider/provider.dart';
+import 'package:seeds/constants/app_colors.dart';
+import 'package:seeds/providers/notifiers/profile_notifier.dart';
+import 'package:seeds/providers/services/eos_service.dart';
+import 'package:seeds/providers/services/navigation_service.dart';
+import 'package:seeds/widgets/main_button.dart';
+import 'package:seeds/widgets/main_text_field.dart';
+
+class Profile extends StatefulWidget {
+  @override
+  _ProfileState createState() => _ProfileState();
+}
+
+class _ProfileState extends State<Profile> {
+  final _nameController = TextEditingController();
+  File _profileImage;
+
+  @override
+  initState() {
+    Future.delayed(Duration.zero).then((_) {
+      ProfileNotifier.of(context).fetchProfile();
+    });
+    super.initState();
+  }
 
-class Profile extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return Center(
-      child: Text("Profile")
+    return Consumer<ProfileNotifier>(
+      builder: (ctx, model, _) {
+        print(
+            'ProfileModel: ${model?.profile?.nickname ?? 'no'}, ${_profileImage}');
+        _nameController.text = model?.profile?.nickname ?? '';
+        return Scaffold(
+          body: ListView(
+            children: <Widget>[
+              Container(
+                width: double.infinity,
+                height: 180.0,
+                decoration: BoxDecoration(
+                  gradient: LinearGradient(colors: AppColors.gradient),
+                ),
+                child: Center(
+                  child: Stack(
+                    children: <Widget>[
+                      GestureDetector(
+                        onTap: () {
+                          if (model?.profile?.image != null)
+                            NavigationService.of(context)
+                                .navigateTo(Routes.imageViewer, model.profile);
+                        },
+                        child: Hero(
+                          tag: 'profilePic',
+                          child: ClipRRect(
+                            borderRadius: BorderRadius.circular(50.0),
+                            child: Container(
+                              width: 100.0,
+                              height: 100.0,
+                              child: (_profileImage != null)
+                                  ? Image.file(
+                                      _profileImage,
+                                      fit: BoxFit.cover,
+                                    )
+                                  : CachedNetworkImage(
+                                      imageUrl: model?.profile?.image ?? '',
+                                      errorWidget: (context, url, error) {
+                                        return Container(
+                                          color: AppColors.getColorByString(
+                                              model?.profile?.nickname ?? ''),
```

Comment:
you are getting the color from nickname?
this needs a comment or some explanation.

Reply:
Yes, I'm getting color from the nickname. I'll add some comments explaining it.